### DISCUSSION TITLE
Best Model Save in Ludwig

### DISCUSSION THREAD

**Discussion by msmmpts** [ludwig-ai/ludwig#3580] [Q&A] [answered]
Hi,

I was running a fine-tuning exercise for Llama 2 model for **10** epochs. Logs however indicated that the best model epoch was 7.

Does the adapter_config.json and adapter_model.bin **correspond to epoch 7**(best epoch) that was reported as the best?
![training_progress](https://github.com/ludwig-ai/ludwig/assets/12732051/f23866ef-9ebb-4445-8835-243246634bc4)

Thanks

**Comment by tgaddair** [comment, marked as answer]
Hey @msmmpts, yes, the final adapter weights that get saved as `model_weights` are taken from the best checkpoint (in this case, epoch 7).

**Reply by msmmpts** [reply]
Thanks
